fix bank initialization

tbg opened the pull request:
the code which computed the initial value of the bank didn't
handle transaction restarts.
removed a redundant transaction in sumAllAccounts.
also print all accounts if the bank messes up.

tamird commented on examples/bank/bank.go:
```diff
@@ -195,23 +194,24 @@ func (bank *Bank) initBankAccounts(cash int64) int64 {
 	}
-	log.Info("done initializing all accounts")
-	return totalCash
+	log.Infof("done initializing all accounts, bank is worth %d$", oldCash+newCash)
```
`$%d`

tschottdorf replied:
done.

tamird commented on examples/bank/bank.go:
```diff
@@ -195,23 +194,24 @@ func (bank *Bank) initBankAccounts(cash int64) int64 {
 func (bank *Bank) periodicallyCheckBalances(expTotalCash int64) {
 	var lastNumTransfers int32
 	lastNow := time.Now()
+	t := time.NewTicker(time.Second)
```
this changes behaviour (continues to tick while the loop below is running). why is that desired?

tschottdorf replied:
you want output every one second, not every `one plus how long the scan takes for that particular iteration`. The output is very jittery. It still is after this change though, because the scan competes with all the transactions on the whole range and needs a lot of restarts.

tamird replied:
sounds reasonable

tamird commented on examples/bank/bank.go:
```diff
@@ -195,23 +194,24 @@ func (bank *Bank) initBankAccounts(cash int64) int64 {
 	var lastNumTransfers int32
 	lastNow := time.Now()
+	t := time.NewTicker(time.Second)
 	for {
```
`for _ := range time.NewTicker(time.Second).C {` if you're gonna keep the ticker

tschottdorf replied:
done.

tamird commented:
LGTM

tamird commented:
LGTM

tbg committed:
fix bank initialization

the code which counted the initial value of the bank didn't
handle transaction restarts.
removed a redundant transaction in sumAllAccounts.
also print all accounts if the bank messes up.

[ci skip]
